java/sage/media/format/FormatParserPlugin.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2015 The SageTV Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sage.media.format;
+
+/**
+ * This class provides an interface for the creation of a plug to override
+ * the built in file format detection.  This will allow the community to create
+ * plugins that will allow SageTV to be able to recognize newer media formates
+ * that the core is not currently able to handle. 
```

Comment:
You might also want to add that it will not override format detection for MP3, image or MPEG2-TS/PS files that the internal format detector can handle.  I'd like to keep it that way, so don't just move where it's invoked in the FormatParser instead. :)  It's really designed to replace the existing FFMPEG format detector.

Reply:
I agree with this completely, and that was my intent.  It looked like a lot of effort was put into handling MPEG2.  I did only want to override the ffmpeg portion.  I will update the comment to try and make that more clear.